script.js

```diff
@@ -1,4 +1,211 @@
+var timesUserWon = 0;
+var timesComputerWon = 0;
+var currentGameMode = "waiting for username";
+var username = "";
+var userWinningPercentage = -1;
+var mostRecentWinner = ``;
+var mostRecentWinnerMessage = ``;
+
+var userWinningOrLosingMessage = function () {
+  if (timesUserWon != 0 && timesComputerWon == 0) {
+    return `Your winning percentage is 100%. Keep is up!`;
+  } else if (timesUserWon == 0 && timesComputerWon == 0) {
+    return `Both have no wins yet.`;
+  } else if (timesUserWon == 0 && timesComputerWon != 0) {
+    return `Your winning percentage is 0%. You are not doing so well.`;
+  } else if (userWinningPercentage < 50) {
+    return `Your winning percentage is ${userWinningPercentage}%. You are not doing so well.`;
+  } else {
+    return `Your winning percentage is ${userWinningPercentage}%. Keep is up!`;
+  }
+};
+
+var checkValidInput = function (input) {
+  if (
+    input != `scissors` &&
+    input != `paper` &&
+    input != `stone` &&
+    input != `reverse`
+  ) {
+    myOutputValue = `There are only 3 options. Try again.`;
+  }
+  return myOutputValue;
+};
+
+var getComputerOutcome = function () {
+  // randomly assigns 1, 2, or 3 to variable 'randomNumber'. function returns one of these strings: 'scissors', 'paper', 'stone'.
+  var randomInteger = Math.floor(Math.random() * 3);
+  randomScissorsPaperStone = ["scissors", "paper", "stone"];
+  return randomScissorsPaperStone[randomInteger];
+};
+
+var displayGameOutcome = function (computerOutcome, userInput) {
+  // outcome: draw
+  if (userInput == computerOutcome) {
+    return `It's a draw.`;
+  } else if (
+    (computerOutcome == `scissors` && userInput == `paper`) ||
+    (computerOutcome == `stone` && userInput == `scissors`) ||
+    (computerOutcome == `paper` && userInput == `stone`)
+  ) {
+    timesComputerWon = timesComputerWon + 1;
+    mostRecentWinner = `computer`;
+    return `You lose.`;
+  } else if (
+    (computerOutcome == `scissors` && userInput == `stone`) ||
+    (computerOutcome == `stone` && userInput == `paper`) ||
+    (computerOutcome == `paper` && userInput == `scissors`)
+  ) {
+    timesUserWon = timesUserWon + 1;
+    mostRecentWinner = `user`;
+    return `You win.`;
+  }
+};
+
+var displayReversedGameOutcome = function (computerOutcome, userInput) {
+  // outcome: draw
+  if (userInput == computerOutcome) {
+    return `It's a draw.`;
+  } else if (
+    (computerOutcome == `paper` && userInput == `scissors`) ||
+    (computerOutcome == `scissors` && userInput == `stone`) ||
+    (computerOutcome == `stone` && userInput == `paper`)
+  ) {
+    timesComputerWon = timesComputerWon + 1;
+    return `You lose.`;
+  } else if (
+    (computerOutcome == `scissors` && userInput == `paper`) ||
+    (computerOutcome == `stone` && userInput == `scissors`) ||
+    (computerOutcome == `paper` && userInput == `stone`)
+  ) {
+    timesUserWon = timesUserWon + 1;
+    return `You win.`;
+  }
```

Comment:
Not so good that we copy-pasted our entire game again, even though we only just changed the winning conditionals. We could have added if statements inside our existing winning conditionals to check for reverse mode and determine a winner there based on the mode, instead of copy-pasting all our code. Same for Korean version